Makefile

```diff
@@ -86,6 +86,8 @@ run: default
 ifeq ("$(wildcard logs/nginx.pid)", "")
 	mkdir -p logs
 	$(OR_EXEC) -p $$PWD/ -c $$PWD/conf/nginx.conf
+else ifeq ("$(shell lsof -p $(shell cat logs/nginx.pid))", "")
+	$(OR_EXEC) -p $$PWD/ -c $$PWD/conf/nginx.conf
```

Comment:
Would be better if we can output the reason to start APISIX even the `nginx.pid` exists.

Reply:
agree +1.

Reply:
Actually we can't...The only thing we know for sure is nginx.pid exists but there's no corresponding process...Also when a user execute `make run`, **the implication** here is that he believes the apisix server is not running, but if it does running, telling him that the server is running, just like the way we do now...But if it doesn't, then we should just start the new server process since the user doesn't care about it...

Reply:
@willmafh 
Yes, our suggestion is to output some message that `nginx.pid exists but there's no corresponding process`, and then start the new one.

Reply:
@spacewander I still don't think it's necessary to output some message here...We can see [bin/apisix](https://github.com/apache/apisix/blob/ce1d40b42f056f46b20386e1937fc81ba69e84c6/bin/apisix#L348) start function, their logic are actually the same...If we really want to output some message here, then the start function of  bin/apisix should also be changed, but it's really unnecessary.

Reply:
@membphis @moonming

Reply:
`nginx.pid exists but there's no corresponding process`

We should add this tip in both `Makefile` and `bin/apisix`, it is more user-friendly.

I created a new issue https://github.com/apache/apisix/issues/2948 , welcome to fix this issue.